doc/whats_new/v0.23.rst

```diff
@@ -185,6 +185,14 @@ Changelog
   `probB_`, are now deprecated as they were not useful. :pr:`15558` by
   `Thomas Fan`_.
 
+- |Fix| Fix use of custom kernel not taking float entries such as string
+  kernels in :class:`svm.SVC` and :class:`svm.SVR`.
+  :pr:`11296` by `Alexandre Gramfort`_ and  :user:`Georgi Peev <georgipeev>`.
+
+- |API| Do not enforce float entries in X when using custom kernel
```

Comment:
We could add this change in `Change of behaviour`. I think this is fine since it could be considered as bug fix but better to document it as well.